server/sonar-server/src/test/java/org/sonar/server/component/ws/SearchProjectsActionTest.java

```diff
@@ -0,0 +1,155 @@
+import static org.sonar.db.component.ComponentTesting.newView;
+import static org.sonar.test.JsonAssert.assertJson;
+
+public class SearchProjectsActionTest {
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+  @Rule
+  public DbTester db = DbTester.create(System2.INSTANCE);
+  ComponentDbTester componentDb = new ComponentDbTester(db);
+  DbClient dbClient = db.getDbClient();
+
+  WsActionTester ws = new WsActionTester(new SearchProjectsAction(dbClient));
+
+  SearchProjectsRequest.Builder request = SearchProjectsRequest.builder();
+
+  @Test
+  public void json_example() {
+    componentDb.insertComponent(newProjectDto()
+      .setUuid(Uuids.UUID_EXAMPLE_01)
+      .setKey(KeyExamples.KEY_PROJECT_EXAMPLE_001)
+      .setName("My Project 1"));
+    componentDb.insertComponent(newProjectDto()
+      .setUuid(Uuids.UUID_EXAMPLE_02)
+      .setKey(KeyExamples.KEY_PROJECT_EXAMPLE_002)
+      .setName("My Project 2"));
+    componentDb.insertComponent(newProjectDto()
+      .setUuid(Uuids.UUID_EXAMPLE_03)
+      .setKey(KeyExamples.KEY_PROJECT_EXAMPLE_003)
+      .setName("My Project 3"));
+
+    String result = ws.newRequest().execute().getInput();
+
+    assertJson(result).withStrictArrayOrder().isSimilarTo(ws.getDef().responseExampleAsString());
+  }
+
+  @Test
+  public void order_by_name() {
+    componentDb.insertComponents(newProjectDto().setName("Maven"), newProjectDto().setName("Apache"), newProjectDto().setName("Guava"));
+
+    SearchProjectsWsResponse result = call(request);
+
+    assertThat(result.getComponentsList()).extracting(Component::getName)
+      .containsExactly("Apache", "Guava", "Maven");
+  }
+
+  @Test
+  public void paginate_result() {
+    IntStream.rangeClosed(1, 9).forEach(i -> componentDb.insertComponent(newProjectDto().setName("PROJECT-" + i)));
+
+    SearchProjectsWsResponse result = call(request.setPage(2).setPageSize(3));
+
+    assertThat(result.getPaging().getPageIndex()).isEqualTo(2);
+    assertThat(result.getPaging().getPageSize()).isEqualTo(3);
+    assertThat(result.getPaging().getTotal()).isEqualTo(9);
+    assertThat(result.getComponentsCount()).isEqualTo(3);
+    assertThat(result.getComponentsList())
+      .extracting(Component::getName)
+      .containsExactly("PROJECT-4", "PROJECT-5", "PROJECT-6");
+  }
+
+  @Test
+  public void return_only_projects() {
+    ComponentDto project = newProjectDto().setName("SonarQube");
+    ComponentDto directory = newDirectory(project, "path");
+    componentDb.insertComponents(project, newModuleDto(project), newView(), newDeveloper("Sonar Developer"), directory, newFileDto(project, directory));
+
+    SearchProjectsWsResponse result = call(request);
+
+    assertThat(result.getComponentsCount()).isEqualTo(1);
+    assertThat(result.getComponents(0).getName()).isEqualTo("SonarQube");
+  }
+
+  @Test
+  public void fail_if_page_size_greater_than_500() {
+    expectedException.expect(IllegalArgumentException.class);
```

Comment:
as this is not testing the error message, there is no guarantee the IAE is throw because of page size